cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -3711,4 +3711,19 @@ String getInternalServiceHostname(String serviceName, boolean useServiceDnsDomai
     protected KafkaStatus createStatus() {
         return new KafkaStatus();
     }
+
+    protected Future<Boolean> delete(Reconciliation reconciliation) {
+        return clusterRoleBindingOperations.reconcile(KafkaResources.initContainerClusterRoleBindingName(reconciliation.name(), reconciliation.namespace()), null)
+                .recover(error -> {
+                    if (error != null
+                            && error.getMessage() != null
+                            && error.getMessage().contains("Message: Forbidden!"))  {
```

Comment:
It it's a `KubernetesClientException` I think there should be a status code, which might be a little less fragile than `contains` on the message.

Reply:
I do not think there is any code which is exposed. This is the logic we use for the creation as well.

Reply:
OK, but I think we should factor out a method for these, both here and for the creation.

Reply:
Hmm, good point. I'm not good at this programming thing ... but I excel in the copy-pasting discipline. And you know how it is ... when all you know is a Cmd-C/Cmd-V everything looks like a copy paste problem :-/. Anyway, it should be fixed now.